tasks/upgrades/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { IDeployments } from '../../scripts/deployment/common'
+
+export const validateDeployments = async (
+  hre: HardhatRuntimeEnvironment,
+  deployments: IDeployments,
+  version: string
+) => {
+  // Check trading lib defined
+  if (!deployments.tradingLib) {
+    throw new Error(
+      `Missing deployed TradingLib for version ${version} in network ${hre.network.name}`
+    )
+  }
+
+  // Check implementations are defined
+  if (
```

Comment:
might be worth splitting these into their own error statements so that we know exactly which ones are missing?

Reply:
I can do the change. I dont think its a big deal to leave it like it is (its just taking a looking at the deployment file and check whats wrong)  but but there is no harm in being more specific. I'll do this.